src/main/java/me/fallenbreath/yamlang/YamlangPlugin.java

```diff
@@ -19,13 +21,24 @@ public void apply(Project project)
 			{
 				String taskName = String.format("yamlangConvert%sResources", StringUtils.capitalize(sourceSet.getName()));
 				project.getLogger().info("Hooking sourceset {}", sourceSet);
-				Task processResources = project.getTasks().getByName(sourceSet.getProcessResourcesTaskName());
 
-				project.getTasks().register(taskName, YamlangConvertor.class, task -> {
-					task.setSourceSet(sourceSet);
-					task.onlyIf(s -> !processResources.getState().getUpToDate());
+				TaskProvider<YamlangConvertResourcesTask> provider = project.getTasks().register(taskName, YamlangConvertResourcesTask.class, task -> {
+					task.dependsOn(sourceSet.getProcessResourcesTaskName());
+					task.getOutputs().upToDateWhen(t -> false);
+
+					DirectoryProperty destination = project.getObjects().directoryProperty().fileValue(sourceSet.getOutput().getResourcesDir());
+					Provider<Directory> inputDir = destination.dir(extension.getInputDir().orElse(""));
```

Comment:
Is it good to give an default value of `""` for a required parameter `inputDir`?

Reply:
The old code does the same, if that's what you mean

Reply:
The `@SkipWhenEmpty` annotation on `YamlangConvertResourcesTask#getInputDirectory`?

Reply:
> The `@SkipWhenEmpty` annotation on `YamlangConvertResourcesTask#getInputDirectory`? -- seems to be

Correct